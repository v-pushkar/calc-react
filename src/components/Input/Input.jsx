import React from "react";

export const Input = ({ type, name, value, onChange, readonly }) => {
  return (
    <div className="input-wrapp" data-testid="input-wrap">
      <input 
      data-testid="input"
        type={type}
        name={name}
        defaultValue={value}        
        // onChange={onChange}
        readOnly={readonly}
      ></input>
    </div>
  );
};
