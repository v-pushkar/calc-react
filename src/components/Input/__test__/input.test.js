import React from "react";
import ReactDOM from "react-dom";
import {getByTestId, render} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import {Input} from "./../Input";
import renderer from "react-test-renderer"

it("Input should be render without crashing",()=>{
    const div = document.createElement("div");
    ReactDOM.render(<Input/>, div)
})

it("Input render correctly",()=>{
    const {getByTestId}= render(<Input  value={"321"}/>)
    expect(getByTestId("input").value).toBe("321")
})

it("marches shapshot input",()=>{
const tree = renderer.create(<Input value={"3215"}></Input>).toJSON()
expect(tree).toMatchSnapshot()
})