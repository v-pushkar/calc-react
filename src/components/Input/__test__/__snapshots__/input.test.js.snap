// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`marches shapshot input 1`] = `
<div
  className="input-wrapp"
  data-testid="input-wrap"
>
  <input
    data-testid="input"
    defaultValue="3215"
  />
</div>
`;
